packages/event-bus-redis/src/services/event-bus-redis.ts

```diff
@@ -53,86 +49,64 @@ type EmitOptions = {
   protected readonly config_: ConfigModule
-  protected readonly manager_: EntityManager
+  protected readonly moduleOptions_: ModuleOptions
+  protected readonly moduleDeclaration_: ConfigurableModuleDeclaration
   protected readonly logger_: Logger
-  protected readonly stagedJobRepository_: typeof StagedJobRepository
-  protected readonly jobSchedulerService_: JobSchedulerService
-  protected readonly eventToSubscribersMap_: Map<
-    string | symbol,
-    SubscriberDescriptor[]
-  >
-  protected readonly redisClient_: Redis.Redis
-  protected readonly redisSubscriber_: Redis.Redis
-  protected queue_: Bull
+  protected readonly stagedJobService_: StagedJobService
+
+  protected queue_: Queue
   protected shouldEnqueuerRun: boolean
-  protected transactionManager_: EntityManager | undefined
   protected enqueue_: Promise<void>
 
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+
   constructor(
-    {
-      manager,
-      logger,
-      stagedJobRepository,
-      redisClient,
-      redisSubscriber,
-      jobSchedulerService,
-    }: InjectedDependencies,
-    config: ConfigModule,
+    { manager, logger, stagedJobService, configModule }: InjectedDependencies,
+    moduleOptions: ModuleOptions = {},
+    moduleDeclaration: ConfigurableModuleDeclaration,
     singleton = true
   ) {
-    this.config_ = config
+    // @ts-ignore
+    super(...arguments)
+
+    if (moduleDeclaration?.resources !== MODULE_RESOURCE_TYPE.SHARED) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_ARGUMENT,
+        "At the moment this module can only be used with shared resources"
+      )
+    }
+
+    this.moduleOptions_ = moduleOptions
+    this.moduleDeclaration_ = moduleDeclaration
+    this.config_ = configModule
     this.manager_ = manager
-    this.logger_ = logger
-    this.jobSchedulerService_ = jobSchedulerService
-    this.stagedJobRepository_ = stagedJobRepository
-
-    if (singleton) {
-      const opts = {
-        createClient: (type: string): Redis.Redis => {
-          switch (type) {
-            case "client":
-              return redisClient
-            case "subscriber":
-              return redisSubscriber
-            default:
-              if (config.projectConfig.redis_url) {
-                return new Redis(config.projectConfig.redis_url)
-              }
-              return redisClient
-          }
-        },
-      }
 
-      this.eventToSubscribersMap_ = new Map()
-      this.queue_ = new Bull(`${this.constructor.name}:queue`, opts)
-      this.redisClient_ = redisClient
-      this.redisSubscriber_ = redisSubscriber
-      // Register our worker to handle emit calls
-      this.queue_.process(this.worker_)
+    this.logger_ = logger
+    this.stagedJobService_ = stagedJobService
 
-      if (process.env.NODE_ENV !== "test") {
-        this.startEnqueuer()
-      }
+    if (singleton && moduleOptions.redisUrl) {
```

Comment:
suggestion: At that moment, if I am right, we have already check that the option redisUrl is present. So it should always be present there no? otherwise we should throw wdyt?

Reply:
I think you are referring to the `redis_url` of the general project config right? That one should be deprecated and eventually removed, as the Redis URL will be passed in the module options going forward.

Reply:
What I mean is that, in the loader, the configuration is checked. Normally, at that point, if the URL is not provided, we would have thrown and the user would already have added it. So I don't thing the check is necessary here right?

Reply:
We are only logging an error when the Redis URL is not passed to avoid crashing the server.

Reply:
It can be an issue then, if the user deploy without having set the redis_url, he does not necessarily look at the logs while deploying, and therefore the connect will not be called and the event bus will throw some error since the queue is not initialized and neither the worker nor the connection. is that correct? as we discuss, it is normally a native feature that the provider restart the previous version if a deployment fail no?

Reply:
Yeah I think that's a fair point. I will add the throwing in the loader 👍